src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/SyncingSubscription.cs

```diff
@@ -0,0 +1,115 @@
+        private long BestSuggestedNumber { get; set; }
+        
+        public SyncingSubscription(IBlockTree blockTree, ILogManager logManager)
+        {
+            _blockTree = blockTree;
+            _logger = logManager.GetClassLogger();
+            
+            IsSyncing = CheckSyncing();
+            if(_logger.IsTrace) _logger.Trace($"Syncing subscription {Id}: Syncing status on start is {IsSyncing}");
+            
+            _blockTree.NewBestSuggestedBlock += OnConditionsChange;
+            if(_logger.IsTrace) _logger.Trace($"Syncing subscription {Id} will track NewBestSuggestedBlocks");
+            
+            _blockTree.NewHeadBlock += OnConditionsChange;
+            if(_logger.IsTrace) _logger.Trace($"Syncing subscription {Id} will track NewHeadBlocks");
+        }
+
+        private bool CheckSyncing()
+        {
+            BestSuggestedNumber = _blockTree.FindBestSuggestedHeader().Number;
+            return BestSuggestedNumber > _blockTree.Head.Number + 8;
+        }
+
+        private void OnConditionsChange(object? sender, BlockEventArgs e)
+        {
+            Task.Run(() =>
+            {
+                bool isSyncing = CheckSyncing();
+
+                if (isSyncing == IsSyncing)
+                {
+                    if (_logger.IsTrace) _logger.Trace($"Syncing subscription {Id} didn't changed syncing status: {IsSyncing}");
+                    return;
+                }
+
+                if (_logger.IsTrace) _logger.Trace($"Syncing subscription {Id} changed syncing status from {IsSyncing} to {isSyncing}");
+
+                IsSyncing = isSyncing;
+
+                JsonRpcResult result = GetJsonRpcResult(isSyncing);
+                JsonRpcDuplexClient.SendJsonRpcResult(result);
+                _logger.Trace($"Syncing subscription {Id} printed SyncingResult object.");
+            }).ContinueWith(
+                t =>
+                    t.Exception?.Handle(ex =>
+                    {
+                        if (_logger.IsDebug) _logger.Debug($"Syncing subscription {Id}: Failed Task.Run.");
+                        return true;
+                    })
+                , TaskContinuationOptions.OnlyOnFaulted
+            );
+        }
+
+        private JsonRpcResult GetJsonRpcResult(bool isSyncing)
+        {
+            JsonRpcResult result =
+                JsonRpcResult.Single(
+                    new JsonRpcSubscriptionResponse()
+                    {
+                        MethodName = "eth_subscribe",
+                        Params = new JsonRpcSubscriptionResult()
+                        {
+                            Subscription = Id,
+                            Result = new SyncingResult
+                            {
+                                IsSyncing = isSyncing,
+                                CurrentBlock = _blockTree.Head.Number,
+                                HighestBlock = BestSuggestedNumber,
+                                StartingBlock = 0L
```

Comment:
This actually doesn't have to be true, but it may be hard to get actual CurrentBlock\StartingBlock\HighestBlock  as there are at least few different cases so lets leave it at that